### ISSUE TITLE
Support `GraphQL`

### ISSUE THREAD

**Issue by heowc** [resilience4j/resilience4j#1573]
Are there any plans to support integration with `graphql-java` (or other graphql implementations) in `resilience4j`?

See https://www.graphql-java.com/

**Comment by heowc** [comment]
Perhaps `resilience4j` can provide a [directive](https://www.graphql-java.com/documentation/master/sdl-directives/) with a name similar to the annotation module by adding the `graphql` module.

**Comment by RobWin** [comment]
Hi, no, there are no plans to create GraphQL directives.
What would be the benefit? Resilience4j should be protocol agnostic and part of your service logic.
If you switch the protocol from GraphQL to REST or gRPC then resilience should still work.

**Comment by heowc** [comment]
My service is using GraphQL. And we are strengthening security by referring to the following [cheat sheet](https://cheatsheetseries.owasp.org/cheatsheets/GraphQL_Cheat_Sheet.html#rate-limiting).

Among them, I looked for several libraries to defend against DoS attacks, and unlike other languages, java did not have a library with related functions.

We decided to use `resilience4j` because it is easy to combine with any library or framework. 👍 However, I said my thoughts because I thought it would be useful for users who are having the same troubles as us.

Of course, maybe someone will depend on resilience4j to make a separate library 🤣
